cwms-data-api/src/main/java/cwms/cda/data/dao/LocationGroupDao.java

```diff
@@ -273,10 +279,21 @@ public List<LocationGroup> getLocationGroups(String groupOfficeId, String catego
                 .on(alcg.LOC_CATEGORY_ID.eq(alga.CATEGORY_ID)
                         .and(alcg.LOC_GROUP_ID.eq(alga.GROUP_ID)));
 
-        if (groupOfficeId != null) {
-            connectBy = onStep.where(DSL.upper(alcg.GRP_DB_OFFICE_ID).eq(groupOfficeId.toUpperCase()).and(condition));
+        if (officeId != null) {
```

Comment:
Seems like these conditions can be simplified. If group office id is not null, then filter group. Else if location office is not null then filter location office. Not sure why there are four branches here

Reply:
Simplified conditioning